npx taze@latest throws [ERR_MODULE_NOT_FOUND]

### Describe the bug

After the latest release I get the following error when running `npx taze@latest`:

```
 ERROR

> @changesets/cli unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> concurrently unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> eslint unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> eslint-plugin-workspaces unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> husky unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> lint-staged unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> playwright unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> prettier unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> turbo unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> typescript unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
> pnpm unknown error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/user/.npm/_npx/8fc85ace19384184/node_modules/semver/functions/valid' imported from /Users/user/.npm/_npx/8fc85ace19384184/node_modules/taze/dist/chunks/index.mjs
Did you mean to import "semver/functions/valid.js"?
```

### Reproduction

Run `npx taze@latest`

### System Info

```Shell
System:
    OS: macOS 14.4.1
    CPU: (8) arm64 Apple M1
    Memory: 136.84 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.12.2 - ~/.asdf/installs/nodejs/20.12.2/bin/node
    npm: 10.5.0 - ~/.asdf/plugins/nodejs/shims/npm
    pnpm: 8.15.7 - ~/.asdf/installs/nodejs/20.12.2/bin/pnpm
  Browsers:
    Chrome: 123.0.6312.124
    Safari: 17.4.1
```


### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/build.config.ts
+++ b/build.config.ts
@@ -7,6 +7,11 @@ export default defineBuildConfig({
   ],
   rollup: {
     inlineDependencies: true,
+    json: {
+      compact: true,
+      namedExports: false,
+      preferConst: true,
+    },
   },
   clean: true,
   declaration: true,

--- a/build.config.ts
+++ b/build.config.ts
@@ -9,8 +9,8 @@ export default defineBuildConfig({
     inlineDependencies: true,
     json: {
       compact: true,
-      namedExports: true,
-      preferConst: true,
+      namedExports: false,
+      preferConst: false,
     },
     commonjs: {
       requireReturnsDefault: 'auto',

--- a/build.config.ts
+++ b/build.config.ts
@@ -9,8 +9,11 @@ export default defineBuildConfig({
     inlineDependencies: true,
     json: {
       compact: true,
-      namedExports: false,
+      namedExports: true,
       preferConst: true,
+    },
+    commonjs: {
+      requireReturnsDefault: 'auto',
     },
     dts: {
       respectExternal: false,

--- a/build.config.ts
+++ b/build.config.ts
@@ -12,6 +12,9 @@ export default defineBuildConfig({
       namedExports: false,
       preferConst: true,
     },
+    dts: {
+      respectExternal: false,
+    },
   },
   clean: true,
   declaration: true,

--- a/package.json
+++ b/package.json
@@ -41,7 +41,6 @@
     "execa": "^8.0.1",
     "picocolors": "^1.0.0",
     "prompts": "^2.4.2",
-    "semver": "^7.6.0",
     "ufo": "^1.5.3",
     "unconfig": "^0.3.13",
     "yargs": "^17.7.2"
@@ -67,6 +66,7 @@
     "npm-package-arg": "^11.0.2",
     "npm-registry-fetch": "^16.2.1",
     "rimraf": "^5.0.5",
+    "semver": "^7.6.0",
     "taze": "workspace:*",
     "typescript": "^5.4.5",
     "unbuild": "^2.0.0",

--- a/package.json
+++ b/package.json
@@ -31,7 +31,7 @@
     "typecheck": "tsc",
     "prepublishOnly": "nr build",
     "release": "bumpp && pnpm publish --no-git-checks",
-    "test": "vitest"
+    "test": "unbuild && vitest"
   },
   "dependencies": {
     "@antfu/ni": "^0.21.12",

--- a/src/cli.ts
+++ b/src/cli.ts
@@ -3,7 +3,7 @@ import type { Argv } from 'yargs'
 import yargs from 'yargs'
 import { hideBin } from 'yargs/helpers'
 import c from 'picocolors'
-import { version } from '../package.json'
+import pkgJson from '../package.json'
 import { check } from './commands/check'
 import { usage } from './commands/usage'
 import { resolveConfig } from './config'
@@ -142,7 +142,7 @@ yargs(hideBin(process.argv))
   )
   .showHelpOnFail(false)
   .alias('h', 'help')
-  .version('version', version)
+  .version('version', pkgJson.version)
   .alias('v', 'version')
   .help()
   .argv

--- a/test/cli.test.ts
+++ b/test/cli.test.ts
@@ -1,0 +1,11 @@
+import path from 'node:path'
+import { expect, it } from 'vitest'
+import { execa } from 'execa'
+
+it('taze cli should just works', async () => {
+  const binPath = path.resolve(__dirname, '../bin/taze.mjs')
+
+  const proc = await execa(process.execPath, [binPath], { stdio: 'pipe' })
+
+  expect(proc.stderr).toBe('')
+})

